### ISSUE TITLE
SuperTux 0.3.5a nonvoid mistake

### ISSUE THREAD

**Issue by nemysis** [SuperTux/supertux#79]
I try to update in openSUSE SuperTux to 0.3.5a, but get in build:

```\n
[  267s] I: Program returns random data in a function
[  267s] E: supertux2 no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/supertux-0.3.5a/src/supertux/screen_manager.hpp:99, 104
[  267s] 
[  267s] I: Program returns random data in a function
[  267s] E: supertux2 no-return-in-nonvoid-function /home/abuild/rpmbuild/BUILD/supertux-0.3.5a/src/supertux/screen_manager.hpp:99, 104
```

Please correct this.

**Comment by maxteufel** [comment]
How do you try to build it (e.g. what tools, exact description of the build procedure) and what's the complete output?

**Comment by nemysis** [comment]
I use build with openSUSE Tools

```\n
osc build
```
and this is

```
cat .build.command
```
```\n
#!/bin/sh -x
exec "rpmbuild" "-ba" "--define" "_srcdefattr (-,root,root)" "--nosignature" "/home/abuild/rpmbuild/SOURCES/supertux2.spec"
```

This is complete output



```\n
[   98s] In file included from /home/abuild/rpmbuild/BUILD/supertux-0.3.5a/src/object/endsequence_fireworks.cpp:21:0:
[   98s] /home/abuild/rpmbuild/BUILD/supertux-0.3.5a/src/supertux/screen_manager.hpp: In member function 'ScreenManager::Action& ScreenManager::Action::operator=(ScreenMana
ger::Action&)':
[   98s] /home/abuild/rpmbuild/BUILD/supertux-0.3.5a/src/supertux/screen_manager.hpp:99:5: warning: no return statement in function returning non-void [-Wreturn-type]
[   98s]      }
[   98s]      ^
[   98s] /home/abuild/rpmbuild/BUILD/supertux-0.3.5a/src/supertux/screen_manager.hpp: In member function 'ScreenManager::Action& ScreenManager::Action::operator=(ScreenMana
ger::Action&&)':
[   98s] /home/abuild/rpmbuild/BUILD/supertux-0.3.5a/src/supertux/screen_manager.hpp:104:5: warning: no return statement in function returning non-void [-Wreturn-type]
[   98s]      }
[   98s]      ^
[   98s] /usr/bin/cmake -E cmake_progress_report /home/abuild/rpmbuild/BUILD/supertux-0.3.5a/build/CMakeFiles 
[   98s] [ 47%] Building CXX object CMakeFiles/supertux2.dir/src/object/weak_block.cpp.o
```

I can't give complete build log here, is not supported. Add please this, not only for images. Thanks

**Comment by maxteufel** [comment]
> I can't give complete build log here, is not supported.

Blame the developers of the software you're using to build it?

> [   98s] /home/abuild/rpmbuild/BUILD/supertux-0.3.5a/src/supertux/screen_manager.hpp:104:5: warning: no return statement in function returning non-void [-Wreturn-type]

This is only a warning, not a fatal build error which would cause a build to fail (Also, the v0.3.5a [CI build](https://travis-ci.org/SuperTux/supertux/builds/61708931) is successful).

It could help if you post the spec file, or even try another one (e.g. the one from fedora).

**Comment by amigadave** [comment]
The complete buld log for supertux 0.3.5a in Fedora is https://kojipkgs.fedoraproject.org//packages/supertux/0.3.5a/1.fc23/data/logs/x86_64/build.log and shows the same warnings (and a few others, but they seem minor). Applying a patch to add "return *this;" to both assignment operators fixes the warning: https://kojipkgs.fedoraproject.org//work/tasks/4350/9954350/build.log

**Comment by maxteufel** [comment]
Oops, forgot to close this when I merged #82.

@amigadave: Try `Closes #XYZ` in the commit message next time so it really closes the issue when it's merged.
